### PR TITLE
core(rrd) move global rrd cache option

### PR DESCRIPTION
**Pull request by loiclau**
# Pull Request Template

## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/class/centreonConfigCentreonBroker.php [hunk 1]
```diff
@@ -213,15 +213,13 @@ public function getTagName($tagId)
         if (!is_null($this->tagsCache) && isset($this->tagsCache[$tagId])) {
             return $this->tagsCache[$tagId];
         }
-        $query = "SELECT tagname
-            FROM cb_tag
-            WHERE cb_tag_id = %d";
+        $query = "SELECT tagname FROM cb_tag WHERE cb_tag_id = %d";
```
Comment (sc979) [inline www/class/centreonConfigCentreonBroker.php]: you could have bound this ;)

Comment (sc979) [reply www/class/centreonConfigCentreonBroker.php]: and the others...

File: www/class/centreonConfigCentreonBroker.php [hunk 2]
```diff
@@ -718,18 +707,14 @@ public function updateConfig($id, $values)
      */
     public function updateCentreonBrokerInfos($id, $values)
     {
-        /*
-        * Clean the informations for this id
-        */
-        $query = "DELETE FROM cfg_centreonbroker_info WHERE config_id = " . $id;
+        // Clean the informations for this id
+        $query = "DELETE FROM WHERE config_id = " . $id;
```
Comment (sc979) [inline www/class/centreonConfigCentreonBroker.php]:
```suggestion
        $query = "DELETE FROM cfg_centreonbroker_info WHERE config_id = " . $id;
```

Comment (sc979) [reply www/class/centreonConfigCentreonBroker.php]: please bound $id and add a PDO transaction / commit / rollback
as you'll insert multiple rows in the foreach occurences... yeahYeah, it's out of your scope

File: www/class/centreonConfigCentreonBroker.php [hunk 3]
```diff
@@ -798,34 +783,32 @@ public function updateCentreonBrokerInfos($id, $values)
...
                             $grp_name = $info[0];
                             $grp_id = $info[1];
-                                        $query = "INSERT INTO cfg_centreonbroker_info
+                            $query = "INSERT INTO cfg_centreonbroker_info
                                         (config_id, config_key, config_value, config_group, config_group_id, grp_level,
                                         subgrp_id, parent_grp_id)
                                         VALUES (" . $id . ", '" . $grp_name . "', '', '" . $group . "', " .
-                                            $gid . ", " . $lvl . ", " . $grp_id . ", " . $parent_id . ")";
-                                        $this->db->query($query);
+                                $gid . ", " . $lvl . ", " . $grp_id . ", " . $parent_id . ")";
```
Comment (sc979) [inline www/class/centreonConfigCentreonBroker.php]: wrong indentation. Follow last line or update last lines indentation

File: www/class/centreonConfigCentreonBroker.php [hunk 4]
```diff
@@ -798,34 +783,32 @@ public function updateCentreonBrokerInfos($id, $values)
...
                         foreach ($fieldvalue as $value) {
-                                       $query = "INSERT INTO cfg_centreonbroker_info
+                            $query = "INSERT INTO cfg_centreonbroker_info
                                         (config_id, config_key, config_value, config_group, config_group_id, grp_level,
                                         subgrp_id, parent_grp_id)
                                         VALUES (" . $id . ", '" . $fieldname . "', '" . $value . "', '" .
-                                           $group . "', " . $gid . ", " . $lvl . ", " .
-                                           $grp_id . ", " . $parent_id . ")";
-                                       $this->db->query($query);
+                                $group . "', " . $gid . ", " . $lvl . ", " .
```
Comment (sc979) [inline www/class/centreonConfigCentreonBroker.php]: wrong indentation. Follow last line or update last lines indentation

File: www/class/centreonConfigCentreonBroker.php [hunk 5]
```diff
@@ -862,7 +845,7 @@ public function getForms($config_id, $tag, $page, $tpl)
         while ($row = $res->fetchRow()) {
```
Comment (sc979) [inline www/class/centreonConfigCentreonBroker.php]: fetchRow

File: www/class/centreonConfigCentreonBroker.php [hunk 6]
```diff
@@ -958,8 +936,8 @@ public function getCorrelationFile()
         $row = $res->fetchRow();
         $configId = $row['config_id'];
         $correlationGroupId = $row['config_group_id'];
-        $query = 'SELECT config_value FROM cfg_centreonbroker_info
-            WHERE config_key = "file" AND config_id = ' . $configId . ' AND config_group_id = ' . $correlationGroupId;
+        $query = 'SELECT config_value FROM cfg_centreonbroker_info  WHERE config_key = "file" AND config_id = '
```
Comment (sc979) [inline www/class/centreonConfigCentreonBroker.php]: for better readability, please send the WHERE clause on next line and the AND clauses on new lines

File: www/include/common/javascript/centreon/brokerConfig.js [hunk 7]
```diff
@@ -141,9 +140,41 @@ jQuery(function () {
...
+var rrdArguments = {
+    // Hook on load tab
+    onLoad: function (element, argument) {
+        argument = window.JSON.parse(argument);
+        return function () {
+            var entry = element.name.match('(input|output)(\\[\\d\\])\\[(\\w*)\\]');
+            var option = entry.input;
+            var target = entry[1] + entry[2] + '[' + argument.target + ']';
+
+            if(document.querySelector('input[name="' + option + '"]:checked').value === 'disable'){
```
Comment (sc979) [inline www/include/common/javascript/centreon/brokerConfig.js]:
```suggestion
            if (document.querySelector('input[name="' + option + '"]:checked').value === 'disable') {
```

File: www/include/common/javascript/centreon/brokerConfig.js [hunk 8]
```diff
@@ -141,9 +140,41 @@ jQuery(function () {
...
+        }
+    },
+    // Hook on change the target
+    onChange: function (argument) {
+        return function (self) {
+            var entry = self.name.match('(input|output)(\\[\\d\\])\\[(\\w*)\\]');
+            var option = entry.input;
+            var target = entry[1] + entry[2] + '[' + argument.target + ']';
+
+            if(document.querySelector('input[name="' + option + '"]:checked').value === 'disable'){
```
Comment (sc979) [inline www/include/common/javascript/centreon/brokerConfig.js]:
```suggestion
            if (document.querySelector('input[name="' + option + '"]:checked').value === 'disable') {
```

File: www/include/common/javascript/centreon/brokerConfig.js (hunk 7 above)
Comment (sc979) [reply www/include/common/javascript/centreon/brokerConfig.js]: just a comment : :checked or .checked

File: www/include/common/javascript/centreon/brokerConfig.js (hunk 8 above)
Comment (sc979) [reply www/include/common/javascript/centreon/brokerConfig.js]: just a comment : :checked or .checked

File: www/install/insertBaseConf.sql [hunk 9]
```diff
@@ -726,7 +730,6 @@ INSERT INTO `cb_type_field_relation` (`cb_type_id`, `cb_field_id`, `is_required`
 (11, 11, 1, 1),
 (11, 12, 1, 2),
 (11, 41, 0, 3),
-(13, 36, 0, 4),
```
Comment (sc979) [inline www/install/insertBaseConf.sql]: just a comment : this one is really hard to find

File: www/install/php/Update-19.10.0-beta.3.php [hunk 10]
```diff
@@ -1 +1,73 @@
...
+if ($cache['value']) {
+    $res = $pearDB->query(
+        "SELECT * FROM cfg_centreonbroker_info WHERE `config_key` = 'type' AND `config_value` = 'rrd'"
+    );
+    $result = $pearDB->query("SELECT `value` FROM options WHERE `key` = 'rrdcached_port' ");
+    $port = $result->fetch();
+
+    while ($row = $res->fetch()) {
+        if ($port['value']) {
+            $query = 'INSERT INTO cfg_centreonbroker_info (config_id, config_key, config_value, '
```
Comment (sc979) [inline www/install/php/Update-19.10.0-beta.3.php]: please bind the variables and remove $query by concatenating the query with the prepared statement

File: www/install/php/Update-19.10.0-beta.3.php [hunk 11]
```diff
@@ -1 +1,73 @@
...
+                . '( ' . $row['config_id'] . ',"rrd_cached_option","tcp",'
+                . $row['config_group'] . ',' . $row['config_group_id'] . ' ),'
+                . '( ' . $row['config_id'] . ',"rrd_cached","' . $port['value'] . '",'
+                . $row['config_group'] . ',' . $row['config_group_id'] . ' )';
+            $pearDB->query($query);
+        } else {
+            $result = $pearDB->query("SELECT `value` FROM options WHERE `key` = 'rrdcached_unix_path' ");
+            $path = $result->fetch();
+
+            $query = 'INSERT INTO cfg_centreonbroker_info (config_id, config_key, config_value, '
```
Comment (sc979) [inline www/install/php/Update-19.10.0-beta.3.php]: please bind the variables and remove $query by concatenating the query with the prepared statement

File: www/install/php/Update-19.10.0-beta.3.php [hunk 12]
```diff
@@ -1 +1,73 @@
...
+ *
+ * For more information : contact@centreon.com
+ *
+ *
+ */
+
+$result = $pearDB->query("SELECT `value` FROM options WHERE `key` = 'rrdcached_enable' ");
+$cache = $result->fetch();
+
+if ($cache['value']) {
```
Comment (sc979) [inline www/install/php/Update-19.10.0-beta.3.php]: couldn't you add a transaction with a rollback and throw an exception in the logs ?

**Comment by sc979** [comment]
yeahYeah, you'll hate me ;)

File: www/install/insertBaseConf.sql (hunk 9 above)
Comment (loiclau) [reply www/install/insertBaseConf.sql]: move on line 850 for add js

File: www/install/php/Update-19.10.0-beta.3.php (hunk 12 above)
Comment (kduret) [reply www/install/php/Update-19.10.0-beta.3.php]: a bit dangerous cause if it fails, nothing appears in frontend, but database is not healthy

File: www/install/sql/centreon/Update-DB-19.10.0-beta.3.sql [hunk 13]
```diff
@@ -4,26 +4,26 @@
...
 
 INSERT INTO `cb_list` (`cb_list_id`, `cb_field_id`, `default_value`)
-VALUES((SELECT coalesce(MAX(l.cb_list_id),0)+1 from cb_list l), (SELECT `cb_field_id` FROM `cb_field` WHERE `description` = 'Use RRD cache.'), 'disable');
+VALUES((SELECT coalesce(MAX(l.cb_list_id),0)+1 from cb_list l), (SELECT `cb_field_id` FROM `cb_field` WHERE `description` = 'Enable rrdcached option for Centreon, please see Centreon documentation to configure it.'), 'disable');
 
 INSERT INTO `cb_list_values` (`cb_list_id`, `value_name`, `value_value`)
 VALUES
 ((SELECT `cb_list_id` FROM `cb_list` WHERE `cb_field_id` =
-  (SELECT `cb_field_id` FROM `cb_field` WHERE `description` = 'Use RRD cache.')), 'Disable', 'disable'
+  (SELECT `cb_field_id` FROM `cb_field` WHERE `description` = 'Enable rrdcached option for Centreon, please see Centreon documentation to configure it.')), 'Disable', 'disable'
```
Comment (kduret) [inline www/install/sql/centreon/Update-DB-19.10.0-beta.3.sql]: should be easier to select using fieldname instead of description :D

**Pull request merged**